From 739698316b0473abda3f0efd3e78dacc4e8bbe8f Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <xiongfeng.wang@linaro.org>
Date: Mon, 8 Jan 2018 07:12:28 -0700
Subject: [PATCH 32/33] fs/hfsplus: use memcpy() instead of strncpy()

gcc-8 reports inlined from 'copy_name' at fs/hfsplus/xattr.c:416:3:
./include/linux/string.h:245:9: warning: '__builtin_strncpy' output
truncated before terminating nul copying 4 bytes from a string of the
same length [-Wstringop-truncation]

Since we don't need to copy '\0', we can just use memcpy() to avoid this
warning.

Signed-off-by: Xiongfeng Wang <xiongfeng.wang@linaro.org>
---
 fs/hfsplus/xattr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/hfsplus/xattr.c b/fs/hfsplus/xattr.c
index f66346155df5..90f60d169217 100644
--- a/fs/hfsplus/xattr.c
+++ b/fs/hfsplus/xattr.c
@@ -241,7 +241,7 @@ static int copy_name(char *buffer, const char *xattr_name, int name_len)
 	int offset = 0;
 
 	if (is_osx_xattr(xattr_name)) {
-		strncpy(buffer, XATTR_MAC_OSX_PREFIX, XATTR_MAC_OSX_PREFIX_LEN);
+		memcpy(buffer, XATTR_MAC_OSX_PREFIX, XATTR_MAC_OSX_PREFIX_LEN);
 		offset += XATTR_MAC_OSX_PREFIX_LEN;
 		len += XATTR_MAC_OSX_PREFIX_LEN;
 	}
-- 
2.22.0

